src/plugins/dashboard/public/dashboard_app/no_data/dashboard_app_no_data.tsx

```diff
@@ -44,6 +61,81 @@ export const DashboardAppNoDataPage = ({
       })
     )
   );
+
+  const lensHelpersAsync = useAsync(() => {
+    return lensService?.stateHelperApi() ?? Promise.resolve(null);
+  }, [lensService]);
+
+  useEffect(() => {
+    return () => {
+      abortController?.abort();
+    };
+  }, [abortController]);
+
+  const onTryESQL = useCallback(async () => {
+    abortController?.abort();
+    if (lensHelpersAsync.value && isMounted()) {
+      const abc = new AbortController();
+      const { dataViews } = dataService;
+      const indexName = (await getIndexForESQLQuery({ dataViews })) ?? '*';
+      const dataView = await getESQLAdHocDataview(`from ${indexName}`, dataViews);
+      const esqlQuery = getInitialESQLQuery(dataView);
+
+      try {
+        const columns = await getESQLQueryColumns({
+          esqlQuery,
+          search: dataService.search.search,
+          signal: abc.signal,
+          timeRange: dataService.query.timefilter.timefilter.getAbsoluteTime(),
+        });
+
+        // lens suggestions api context
+        const context = {
+          dataViewSpec: dataView?.toSpec(false),
+          fieldName: '',
+          textBasedColumns: columns,
+          query: { esql: esqlQuery },
+        };
+
+        setAbortController(abc);
+
+        const chartSuggestions = lensHelpersAsync.value.suggestions(context, dataView);
+        if (chartSuggestions?.length) {
+          const [suggestion] = chartSuggestions;
+
+          const attrs = getLensAttributesFromSuggestion({
+            filters: [],
+            query: {
+              esql: esqlQuery,
+            },
+            suggestion,
+            dataView,
+          }) as TypedLensByValueInput['attributes'];
+
+          const lensEmbeddableInput = {
+            attributes: attrs,
+            id: generateId(),
+          };
+
+          await embeddableService.getStateTransfer().navigateToWithEmbeddablePackage('dashboards', {
+            state: {
+              type: 'lens',
+              input: lensEmbeddableInput,
+            },
+            path: '#/create',
+          });
+        }
+      } catch (error) {
+        coreServices.notifications.toasts.addWarning(
```

Comment:
Toast should not be displayed if error is aborted request

Reply:
Good catch, done https://github.com/elastic/kibana/pull/196658/commits/a6da3da6dc5eb3b712d1d94c1d1d9239b5761414